src/org/opendatakit/briefcase/ui/reused/UiFieldValidator.java

```diff
@@ -0,0 +1,74 @@
+package org.opendatakit.briefcase.ui.reused;
+
+import java.awt.Color;
+import java.awt.event.KeyAdapter;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+import javax.swing.JLabel;
+import javax.swing.JTextField;
+import org.opendatakit.briefcase.reused.http.RequestBuilder;
+
+public class UiFieldValidator {
+  public static final Predicate<String> REQUIRED = s -> s != null && !s.trim().isEmpty();
+  public static final Predicate<String> NUMBER = s -> s == null || s.matches("\\d+");
+  public static final Predicate<String> EMAIL = s -> s == null || s.matches("\\S+@\\S+");
+  public static final Predicate<String> URI = s -> s == null || RequestBuilder.isUri(s);
+  static final Color COLOR_NOT_VALID = new Color(215, 25, 28); // Color-blind safe red
```

Comment:
Did you consider INVALID instead of NOT_VALID? (There is the potential that in-'val-id could be confused with 'in-vuh-lid, a person made weak or disabled by illness or injury.)

Reply:
Yes, that's precisely why I went for NOT_VALID. In this case, we're using colors adapted for people with different eyesight and it feels even more important to avoid the "invalid" term.

Reply:
I feel like this might be a bigger deal in Spanish than in English. Do you think it's OK to use "invalid"?

Reply:
I think, here in the USA at least, the risk of misunderstanding and upset is very small, probably not justifying avoiding the natural antonym. I’m cool with it either way.